gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -121,39 +121,118 @@ cmd void vkDestroyFramebuffer(
 // Renderpass //
 ////////////////
 
-@internal class SubpassDescription {
-  @unused VkSubpassDescriptionFlags        Flags
-  @unused VkPipelineBindPoint              PipelineBindPoint
-  @unused map!(u32, VkAttachmentReference) InputAttachments
-  @unused map!(u32, VkAttachmentReference) ColorAttachments
-  @unused map!(u32, VkAttachmentReference) ResolveAttachments
-  @unused ref!VkAttachmentReference        DepthStencilAttachment
-  @unused map!(u32, u32)                   PreserveAttachments
+@internal enum RenderPassVersion : u32 {
+  Renderpass = 0,
+  Renderpass2 = 1,
+  Renderpass2KHR = 2,
 }
 
 @internal class RenderPassObject {
   @unused VkDevice                           Device
   @unused VkRenderPass                       VulkanHandle
-  @unused map!(u32, VkAttachmentDescription) AttachmentDescriptions
-  @unused map!(u32, SubpassDescription)      SubpassDescriptions
-  @unused map!(u32, VkSubpassDependency)     SubpassDependencies
+  @unused map!(u32, AttachmentDescription2)  AttachmentDescriptions
+  @unused map!(u32, SubpassDescription2)     SubpassDescriptions
+  @unused map!(u32, SubpassDependency2)      SubpassDependencies
```

Comment:
`AttachmentDescription2`, `SubpassDescription2`, and `SubpassDependency2` are all "internal" objects, so please drop the `2`. Also keep them here rather than moving them to the extension file. If you want to can annotate the `ViewMask` field to be 1.2 only.

Reply:
I believe this is not a very good idea. Because we have exactly a `VkSubpassDescription2` structure that corresponds to the internal structure while the `VkSubpassDescription` does not reflect it.

We will have more of these in the future with Vulkan creating more of these and I would like to keep consistency.

If you want, we can say that there is no reflection between the spec struct and internal Struct and in that case I am ok with something like `VkSubpassDescriptionX` as well.

Reply:
@pmuetschard

Reply:
The difference between `vk<Something>` and `vk<something>2` is usually just the addition of the `pNext` pointer, which we don't put into our internal structs anyways. There's not going to be many more `vk<something>2` as there's not much, if anything, left that doesn't have a `pNext`, there certainly won't be any `vk<something>3` - Khronos have learnt their lesson. I'd rather not follow this crazy `<something>2` pattern, especially since we do not have the `<something>` beside it. It'd be a different story if we had both. Also, when extensions show up that add structs to the `pNext` chain, we will adjust the `<something>` structs and add fields to them that are not part of the `vk<something>` and so they won't match anymore.

Reply:
Please see the `rename internal structs` commit. Done!